tests/foreman/maintain/test_packages.py

```diff
@@ -0,0 +1,318 @@
+    logger.info(result.stdout)
+    assert 'Packages are locked' in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.execute('dnf install -y zsh')
+    assert result.status == 1
+    assert 'Use foreman-maintain packages install/update <package>' in result.stdout
+
+    # Test whether satellite-maintain packages install/ update command works as expected.
+    result = sat_maintain.cli.Packages.install(packages='zsh', options={'assumeyes': True})
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'Nothing to do' not in result.stdout
+    assert 'Packages are locked.' in result.stdout
+    assert 'Automatic locking of package versions is enabled in installer.' in result.stdout
+
+    result = sat_maintain.cli.Packages.update(packages='zsh', options={'assumeyes': True})
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'Nothing to do' not in result.stdout
+    assert 'Packages are locked.' in result.stdout
+    assert 'Automatic locking of package versions is enabled in installer.' in result.stdout
+
+    # Test whether packages are unlocked or not
+    result = sat_maintain.install(InstallerCommand('no-lock-package-versions'))
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'Success!' in result.stdout
+
+    result = sat_maintain.cli.Packages.status()
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'Packages are not locked.' in result.stdout
+    assert 'Automatic locking of package versions is disabled in installer.' in result.stdout
+
+    result = sat_maintain.cli.Packages.is_locked()
+    logger.info(result.stdout)
+    assert 'Packages are not locked' in result.stdout
+    assert result.status == 1
+
+    result = sat_maintain.execute('dnf remove -y zsh')
+    assert result.status == 0
+
+    result = sat_maintain.execute('dnf install -y zsh')
+    assert result.status == 0
+    assert 'Use foreman-maintain packages install/update <package>' not in result.stdout
+
+    @request.addfinalizer
+    def _finalize():
+        assert sat_maintain.execute('dnf remove -y zsh').status == 0
+        if sat_maintain.__class__.__name__ == 'Satellite':
+            result = sat_maintain.install(InstallerCommand('lock-package-versions'))
+            logger.info(result.stdout)
+            assert result.status == 0
+            assert 'Success!' in result.stdout
+
+
+@pytest.mark.include_capsule
+def test_positive_fm_packages_update(request, sat_maintain):
+    """Verify whether packages check-update and update work as expected.
+
+    :id: 354d9940-10f1-4244-9079-fdbd24be49b3
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain packages check-update
+        2. Run satellite-maintain packages update to update the walrus package.
+
+    :BZ: 1803975
```

Comment:
I see a customer case with this BZ, better to add `:customerscenario: true`

Reply:
@shubhamsg199 updated